### PR TITLE
Washington's updates to terrame 2.0-RC-7 - part 4

### PR DESCRIPTION
**Pull request by wsenafranca**
- [x] #2133 
- [x] #2144 
- [x] #2140
- [x] #2044 
- [x] #2152

### CODE REVIEW

File: build/cmake/find/FindGMock.cmake [hunk 1]
```diff
@@ -60,71 +60,70 @@
...
 if(MSVC AND GMOCK_MSVC_SEARCH STREQUAL "MD")
-  # The provided /MD project files for Google Mock add -md suffixes to the
-  # library names.
-  _gmock_find_library(GMOCK_LIBRARY            gmock-md  gmock)
-  _gmock_find_library(GMOCK_LIBRARY_DEBUG      gmock-mdd gmockd)
-  _gmock_find_library(GMOCK_MAIN_LIBRARY       gmock_main-md  gmock_main)
-  _gmock_find_library(GMOCK_MAIN_LIBRARY_DEBUG gmock_main-mdd gmock_maind)
+	# The provided /MD project files for Google Mock add -md suffixes to the
+	# library names.
+	_gmock_find_library(GMOCK_LIBRARY				gmock-md  gmock)
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindGMock.cmake]: The column with the items gmock-md... here should be aligned in the same way it was before.

File: build/cmake/find/FindGMock.cmake [hunk 2]
```diff
@@ -60,71 +60,70 @@
...
+	_gmock_find_library(GMOCK_LIBRARY_DEBUG			gmock-mdd gmockd)
+	_gmock_find_library(GMOCK_MAIN_LIBRARY			gmock_main-md  gmock_main)
+	_gmock_find_library(GMOCK_MAIN_LIBRARY_DEBUG	gmock_main-mdd gmock_maind)
 else()
-  _gmock_find_library(GMOCK_LIBRARY            gmock)
-  _gmock_find_library(GMOCK_LIBRARY_DEBUG      gmockd)
-  _gmock_find_library(GMOCK_MAIN_LIBRARY       gmock_main)
-  _gmock_find_library(GMOCK_MAIN_LIBRARY_DEBUG gmock_maind)
+	_gmock_find_library(GMOCK_LIBRARY				gmock)
+	_gmock_find_library(GMOCK_LIBRARY_DEBUG			gmockd)
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindGMock.cmake]: Same here.

File: build/cmake/find/FindQtLua-extras.cmake [hunk 3]
```diff
@@ -30,24 +30,22 @@ cmake_minimum_required(VERSION 3.0)
...
 find_library(QTLUAEXTRAS_LIBRARY
-   NAMES qtluae
-   PATHS 	/usr/lib 
-			/usr/local/lib 
-			/opt/lib 
-			/opt/local/lib 
-			/usr/local/qtlua-extras/lib
-			${TERRAME_DEPENDENCIES_DIR}/lib
-)
+			 NAMES qtluae
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindQtLua-extras.cmake]: Too much indentation here.

File: build/cmake/find/FindQtLua.cmake [hunk 4]
```diff
@@ -30,19 +30,17 @@ cmake_minimum_required(VERSION 3.0)
 # Find library - - tries to find *.a,*.so,*.dylib in paths hard-coded by the script
 
 find_library(QTLUA_LIBRARY
-   NAMES qtlua
-   PATHS /usr/lib /usr/local/lib /opt/lib /opt/local/lib /usr/local/qtlua/lib
-)
+			 NAMES qtlua
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindQtLua.cmake]: Too much here as well. use only one additional tab when defining parameters of a functions in separate lines.

File: build/cmake/find/FindQwt.cmake [hunk 5]
```diff
@@ -30,46 +30,46 @@
...
-  find_path(QWT_INCLUDE_DIR qwt.h
-            PATHS /usr
-                  /usr/local
-                  /usr/local/qwt
-            PATH_SUFFIXES include
-                          qwt
-                          include/qwt
-                          lib/qwt.framework/Headers)
+	find_path(QWT_INCLUDE_DIR qwt.h
+			  PATHS /usr
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindQwt.cmake]: Note the space before PATHS. It should not exist. Also, use only one additional tab, it seems that you are using two here.

File: build/cmake/terrame-macros.cmake [hunk 6]
```diff
@@ -79,34 +76,33 @@ ENDMACRO(TmeInstallQt5Plugins)
...
-  foreach(_plugin_dir ${_plugin_dirs})
+	set(_plugin_dirs "imageformats;iconengines;platforms") #TODO(#1711)
 
-    install (DIRECTORY ${QT_PLUGINS_DIR}/${_plugin_dir}
-             DESTINATION ${_dest}  COMPONENT runtime
-             FILES_MATCHING REGEX "${_regex_exp}")
-  endforeach()
+	foreach(_plugin_dir ${_plugin_dirs})
+		install(DIRECTORY ${QT_PLUGINS_DIR}/${_plugin_dir}
+				DESTINATION ${_dest}  COMPONENT runtime
```
Comment (pedro-andrade-inpe) [inline build/cmake/terrame-macros.cmake]: Two tabs here, should be one.

**Comment by pedro-andrade-inpe** [comment]
@avancinirodrigo, review the PR after @wsenafranca commit after my suggestions.

File: build/cmake/find/FindGMock.cmake [hunk 7]
```diff
@@ -60,71 +60,67 @@
...
+		msvc/gmock/Release)
+	endif()
 endif()
 
 find_path(GMOCK_INCLUDE_DIR gmock/gmock.h
-  HINTS
-    $ENV{GMOCK_ROOT}/include
-    ${GMOCK_ROOT}/include
-)
+          HINTS $ENV{GMOCK_ROOT}/include
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindGMock.cmake]: Too much indentation here.

File: build/cmake/find/FindQtLua-extras.cmake [hunk 8]
```diff
@@ -30,24 +30,22 @@ cmake_minimum_required(VERSION 3.0)
...
 find_library(QTLUAEXTRAS_LIBRARY
-   NAMES qtluae
-   PATHS 	/usr/lib 
-			/usr/local/lib 
-			/opt/lib 
-			/opt/local/lib 
-			/usr/local/qtlua-extras/lib
-			${TERRAME_DEPENDENCIES_DIR}/lib
-)
+             NAMES qtluae
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindQtLua-extras.cmake]: Here too.

File: build/cmake/find/FindQtLua-extras.cmake [hunk 9]
```diff
@@ -30,24 +30,22 @@ cmake_minimum_required(VERSION 3.0)
...
 # Export include and library path for linking with other libraries
 # Find path - tries to find *.h in paths hard-coded by the script
 find_path(QTLUAEXTRAS_INCLUDE_DIR qtluae_version.hpp
-	HINTS  	/usr/include/QtLuaExtras 
-			/usr/local/include/QtLuaExtras 
-			/usr/local/qtlua-extras/include/QtLuaExtras
-			${TERRAME_DEPENDENCIES_DIR}/include/QtLuaExtras
-			${TERRALIB_3RDPARTY_DIR}/include/QtLuaExtras
-)
+          HINTS /usr/include/QtLuaExtras
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindQtLua-extras.cmake]: And here. Use tabs.

File: build/cmake/find/FindQtLua.cmake [hunk 10]
```diff
@@ -30,19 +30,17 @@ cmake_minimum_required(VERSION 3.0)
 # Find library - - tries to find *.a,*.so,*.dylib in paths hard-coded by the script
 
 find_library(QTLUA_LIBRARY
-   NAMES qtlua
-   PATHS /usr/lib /usr/local/lib /opt/lib /opt/local/lib /usr/local/qtlua/lib
-)
+             NAMES qtlua
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindQtLua.cmake]: Here.

File: build/cmake/find/FindQtLua.cmake [hunk 11]
```diff
@@ -30,19 +30,17 @@ cmake_minimum_required(VERSION 3.0)
...
 # Export include and library path for linking with other libraries
 # Find path - tries to find *.h in paths hard-coded by the script
 find_path(QTLUA_INCLUDE_DIR qtluafunction.hh
-	HINTS  	/usr/include/QtLua 
-			/usr/local/include/QtLua 
-			/usr/local/qtlua/include/QtLua
-			${TERRAME_DEPENDENCIES_DIR}/include/QtLua
-			${TERRALIB_3RDPARTY_DIR}/include/QtLua
-)
+          HINTS /usr/include/QtLua
```
Comment (pedro-andrade-inpe) [inline build/cmake/find/FindQtLua.cmake]: Here.

File: build/cmake/terrame/CMakeLists.txt [hunk 12]
```diff
@@ -332,425 +328,371 @@ endif()
 		#)
 
 if (APPLE)
-    if (TERRAME_BUILD_AS_BUNDLE)
+	if (TERRAME_BUILD_AS_BUNDLE)
```
Comment (pedro-andrade-inpe) [inline build/cmake/terrame/CMakeLists.txt]: remove empty space after if.

File: build/cmake/terrame/CMakeLists.txt [hunk 13]
```diff
@@ -332,425 +328,371 @@ endif()
...
+	                              ${MSYS_DIR}/md5sum.exe
+	                              ${MSYS_DIR}/mktemp.exe
+	                              ${MSYS_DIR}/msys-1.0.dll
+	                              ${MSYS_DIR}/msys-iconv-2.dll
+	                              ${MSYS_DIR}/msys-intl-8.dll
+	                              ${MSYS_DIR}/rm.exe
+	                              ${MSYS_DIR}/touch.exe
+	                              ${MSYS_DIR}/unzip.exe
+	                              ${MSYS_DIR}/zip.exe
+	                              ${MSYS_DIR}/mv.exe
```
Comment (avancinirodrigo) [inline build/cmake/terrame/CMakeLists.txt]: Join this `)` to last line.

File: build/cmake/terrame/CMakeLists.txt [hunk 14]
```diff
@@ -332,425 +328,371 @@ endif()
...
-			)
+	        DESTINATION bin/share/terralib/plugins
+	        COMPONENT runtime)
 
 	install(DIRECTORY ${TERRALIB_DIR}/share/terralib/schemas/
-			DESTINATION bin/share/terralib/schemas
-			COMPONENT runtime
-			)
+	        DESTINATION bin/share/terralib/schemas
+	        COMPONENT runtime)
```
Comment (avancinirodrigo) [inline build/cmake/terrame/CMakeLists.txt]: Remove extra space before that.

File: build/cmake/testing/terrame/inttest/CMakeLists.txt [hunk 15]
```diff
@@ -84,29 +84,32 @@ file(GLOB TERRAME_INTTEST_CORE_HDR_FILES ${TERRAME_ABSOLUTE_ROOT_DIR}/inttest/co
...
+                       ${TERRAME_OBSERVER_TYPES_CHART_PLOT_SRC_FILES} ${TERRAME_OBSERVER_TYPES_CHART_PLOT_HDR_FILES}
+                       ${TERRAME_OBSERVER_UDP_SENDER_SRC_FILES} ${TERRAME_OBSERVER_UDP_SENDER_HDR_FILES}
+                       ${TERRAME_GEN_HDR_FILES})
+
+target_link_libraries(inttest ${GTEST_BOTH_LIBRARIES} ${GMOCK_BOTH_LIBRARIES}
+                              ${Qt5Network_LIBRARIES} ${Qt5Core_LIBRARIES} ${Qt5Gui_LIBRARIES} ${Qt5Widgets_LIBRARIES}
+                              ${QWT_LIBRARY})
 
 install(TARGETS inttest
 		RUNTIME DESTINATION bin
```
Comment (avancinirodrigo) [inline build/cmake/testing/terrame/inttest/CMakeLists.txt]: Decrease one tab.

File: build/cmake/testing/terrame/unittest/CMakeLists.txt [hunk 16]
```diff
@@ -44,7 +44,7 @@ source_group("Source Files\\core" FILES ${TERRAME_UNITTEST_CORE_SRC_FILES})
 source_group("Header Files\\core" FILES ${TERRAME_UNITTEST_CORE_HDR_FILES})
 
 add_executable(unittest ${TERRAME_UNITTEST_SRC_FILES}
-						${TERRAME_UNITTEST_CORE_SRC_FILES} ${TERRAME_UNITTEST_CORE_HDR_FILES})
+                        ${TERRAME_UNITTEST_CORE_SRC_FILES} ${TERRAME_UNITTEST_CORE_HDR_FILES})
 
 target_link_libraries(unittest ${GTEST_BOTH_LIBRARIES} ${GMOCK_BOTH_LIBRARIES})
```
Comment (avancinirodrigo) [inline build/cmake/testing/terrame/unittest/CMakeLists.txt]: Decrease one tab in `RUNTIME` and `COMPONENT`.

**Comment by pedro-andrade-inpe** [comment]
@avancinirodrigo, please review the changes here again.

File: build/cmake/find/FindGMock.cmake [hunk 17]
```diff
@@ -85,19 +85,20 @@ endif()
...
+			msvc/gmock/Debug
+			msvc/gmock/Release)
 	endif()
 endif()
 
 find_path(GMOCK_INCLUDE_DIR gmock/gmock.h
-          HINTS $ENV{GMOCK_ROOT}/include
-                ${GMOCK_ROOT}/include)
+	HINTS 
+		$ENV{GMOCK_ROOT}/include
```
Comment (avancinirodrigo) [inline build/cmake/find/FindGMock.cmake]: Change to:
```cmake
find_path(GMOCK_INCLUDE_DIR gmock/gmock.h HINTS $ENV{GMOCK_ROOT}/include ${GMOCK_ROOT}/include)
```

File: build/cmake/find/FindGMock.cmake [hunk 18]
```diff
@@ -85,19 +85,20 @@ endif()
...
+		${GMOCK_ROOT}/include)
 
 mark_as_advanced(GMOCK_INCLUDE_DIR)
```
Comment (avancinirodrigo) [inline build/cmake/find/FindGMock.cmake]: Remove extra spaces line 124:
```cmake
_gmock_append_debugs(GMOCK_LIBRARIES #HERE# GMOCK_LIBRARY)
```

File: build/cmake/find/FindQtLua-extras.cmake [hunk 19]
```diff
@@ -30,22 +30,22 @@ cmake_minimum_required(VERSION 3.0)
...
 
 find_library(QTLUAEXTRAS_LIBRARY
-             NAMES qtluae
-             PATHS /usr/lib 
-                   /usr/local/lib
-                   /opt/lib 
-                   /opt/local/lib 
-                   /usr/local/qtlua-extras/lib
-                   ${TERRAME_DEPENDENCIES_DIR}/lib)
+	NAMES qtluae
```
Comment (avancinirodrigo) [inline build/cmake/find/FindQtLua-extras.cmake]: Put `NAMES` to up together QTLUAEXTRAS_LIBRARY.

File: build/cmake/find/FindQtLua.cmake [hunk 20]
```diff
@@ -30,17 +30,17 @@ cmake_minimum_required(VERSION 3.0)
 # Find library - - tries to find *.a,*.so,*.dylib in paths hard-coded by the script
 
 find_library(QTLUA_LIBRARY
-             NAMES qtlua
-             PATHS /usr/lib /usr/local/lib /opt/lib /opt/local/lib /usr/local/qtlua/lib)
+	NAMES qtlua
```
Comment (avancinirodrigo) [inline build/cmake/find/FindQtLua.cmake]: Move `NAMES` to up.

File: build/cmake/terrame/CMakeLists.txt (hunk 14 above)
Comment (avancinirodrigo) [reply build/cmake/terrame/CMakeLists.txt]: Sorry, I mean extra line.

**Pull request merged**